armory/__main__.py

```diff
@@ -211,6 +201,92 @@ def clean(command_args, prog, description):
                     raise
 
 
+def _get_path(name, default_path, absolute_required=True):
+    answer = None
+    while answer is None:
+        try:
+            answer = input(f'{name} [DEFAULT: "{default_path}"]: ')
+        except EOFError:
+            answer = ""
+        if not answer:
+            answer = default_path
+        answer = os.path.expanduser(answer)
+        if os.path.isabs(answer):
+            answer = os.path.abspath(answer)
+        elif absolute_required:
+            print(f"Invalid answer: '{answer}' Absolute path required for {name}")
+            answer = None
+        else:
+            answer = os.path.relpath(answer)
+    return answer
+
+
+def configure(command_args, prog, description):
+    parser = argparse.ArgumentParser(prog=prog, description=description)
+    parser.add_argument(
+        "-d",
+        "--debug",
+        dest="log_level",
+        action="store_const",
+        const=logging.DEBUG,
+        default=logging.INFO,
+        help="Debug output (logging=DEBUG)",
+    )
+    args = parser.parse_args(command_args)
+    coloredlogs.install(level=args.log_level)
+
+    default = paths.default()
+
+    instructions = "\n".join(
+        [
+            "Configuring paths for armory usage",
+            f'    This configuration will be stored at "{default.armory_config}"',
+            "",
+            "Please enter desired target directory for the following paths.",
+            "    If left empty, the default path will be used.",
+            "    Absolute paths (which include '~' user paths) are required.",
+            "",
+        ]
+    )
+    print(instructions)
+
+    config = {
+        "dataset_dir": _get_path("datasets", default.dataset_dir),
+        "model_dir": _get_path("models", default.model_dir),
+        "tmp_dir": _get_path("tmp", default.tmp_dir),
+        "output_dir": _get_path("output", default.output_dir),
+    }
+    resolved = "\n".join(
+        [
+            "Resolved paths:",
+            f"    datasets: {config['dataset_dir']}",
+            f"    models:   {config['model_dir']}",
+            f"    tmp:      {config['tmp_dir']}",
+            f"    output:   {config['output_dir']}",
+            "",
+        ]
+    )
+    print(resolved)
+    save = None
+    while save is None:
+        if os.path.isfile(default.armory_config):
+            print("WARNING: this will overwrite existing configuration.")
+            print("    Press Ctrl-C to abort.")
+        answer = input("Save this configuration? [y/N] ")
+        if answer in ("Y", "y"):
+            print("Saving configuration...")
+            paths.save_config(config)
+            print("Configure successful")
+            save = True
+        elif answer in ("N", "n"):
+            print("Configuration not saved")
+            save = False
+        else:
+            print(f"Invalid selection: {answer}")
+        print()
```

Comment:
So the capital N tells me this should be the default configuration if I just hit enter. Also should we default it to Y?

Reply:
Oh, that's what the capital is about. Yeah, that makes sense - defaulting to Y.